assets/js/controllers/sfox/sfoxVerify.controller.js

```diff
@@ -2,10 +2,20 @@ angular
   .module('walletApp')
   .controller('SfoxVerifyController', SfoxVerifyController);
 
-function SfoxVerifyController ($rootScope, $scope, $q, state, $http, sfox, modals, Upload, QA) {
-  $scope.states = state.stateCodes;
+function SfoxVerifyController ($rootScope, $scope, $q, state, $http, sfox, modals, Upload, QA, Options) {
   let exchange = $scope.vm.exchange;
 
+  let processOptions = (options) => {
+    let states = options.partners.sfox.states;
+    $scope.states = state.stateCodes.filter((s) => states.indexOf(s.Code) > -1);
+  };
+
+  if (Options.didFetch) {
+    processOptions(Options.options);
+  } else {
+    Options.get().then(processOptions);
```

Comment:
`Options.get()` already checks for `didFetch`. Could you resolve the promise in routes.js so that you can process it immediately in the controller?